lib/ui/views/installer/installer_viewmodel.dart

```diff
@@ -169,6 +179,72 @@ class InstallerViewModel extends BaseViewModel {
     }
   }
 
+  Future<void> copyLogs() async {
+    final info = await AboutInfo.getInfo();
+
+    final formattedLogs = [
+      '```',
+      '~ Device Info',
+      'ReVanced Manager: ${info['version']}',
+      'Build: ${info['flavor']}',
+      'Model: ${info['model']}',
+      'Android version: ${info['androidVersion']}',
+      'Supported architectures: ${info['supportedArch'].join(", ")}',
+
+      '\n~ Patch Info',
+      'App: ${_app.packageName} v${_app.version}',
+      'Patches version: ${_managerAPI.patchesVersion}',
+      'Patches: ${_patches.map((p) => p.name).toList().join(", ")}',
+
+      '\n~ Settings',
+      'Enabled changing patches: ${_managerAPI.isPatchesChangeEnabled()}',
+      'Enabled universal patches: ${_managerAPI.areUniversalPatchesEnabled()}',
+      'Enabled experimental patches: ${_managerAPI.areExperimentalPatchesEnabled()}',
+      'Patches source: ${_managerAPI.getPatchesRepo()}',
+      'Integration source: ${_managerAPI.getIntegrationsRepo()}',
+
+      '\n~ Logs',
+      logs,
+      '```',
+    ];
+
+    Clipboard.setData(ClipboardData(text: formattedLogs.join('\n')));
+    _toast.showBottom('installerView.copiedToClipboard');
+  }
+
+  Future<void> screenshotDetected(BuildContext context) async {
+    await showDialog(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: I18nText(
+          'warning',
+        ),
+        backgroundColor: Theme.of(context).colorScheme.secondaryContainer,
+        icon: const Icon(Icons.warning),
+        content: SingleChildScrollView(
+          child: I18nText('installerView.screenshotDetected'),
+        ),
+        actions: <Widget>[
+          CustomMaterialButton(
+            isFilled: false,
+            label: I18nText('noButton'),
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+          ),
+          CustomMaterialButton(
+            label: I18nText('yesButton'),
+            onPressed: () {
+              copyLogs();
+              showPopupScreenshotWarning = true;
```

Comment:
Is this intended? After the screenshot event handler was called, it should not be called for the current installer session anymore, so that the user can copy a screenshot without any intervention.

Reply:
It was intended because if the user wants to copy logs then receives another error,  they know that screenshoting will open the dialog to copy logs, so that is what they do. They may not know that there is a dedicated button outside the dialog so I programmed it this way to prevent confusion. If the user does not want the popup popping up, they can simply press no.